dashboards-observability/public/components/visualizations/charts/box_plot/box_plot.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { isEmpty, last } from 'lodash';
+import { Plt } from '../../plotly/plot';
+import { EmptyPlaceholder } from '../../../event_analytics/explorer/visualizations/shared_components/empty_placeholder';
+import { AvailabilityUnitType } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_availability';
+import { ThresholdUnitType } from '../../../event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_thresholds';
+import { ConfigListEntry } from '../../../../../common/types/explorer';
+import {
+  hexToRgb,
+  filterDataConfigParameter,
+  getTooltipHoverInfo,
+} from '../../../event_analytics/utils/utils';
+import { FILLOPACITY_DIV_FACTOR } from '../../../../../common/constants/shared';
+import {
+  LONG_CHART_COLOR,
+  PLOTLY_COLOR,
+  THRESHOLD_LINE_WIDTH,
+  THRESHOLD_LINE_OPACITY,
+  MAX_BUCKET_LENGTH,
+} from '../../../../../common/constants/shared';
+import { DefaultBoxChartStyles } from '../../../../../common/constants/explorer';
+
+const { PointPosition, MarkerLineWidth } = DefaultBoxChartStyles;
+export const BoxPlot = ({ visualizations, layout, config }: any) => {
+  const { vis } = visualizations;
+  const {
+    data,
+    metadata: { fields },
+  } = visualizations.data.rawVizData;
+  const lastIndex = fields.length - 1;
+  const {
+    dataConfig: {
+      chartStyles = {},
+      valueOptions = {},
+      legend = {},
+      colorTheme = [],
+      panelOptions = {},
+      tooltipOptions = {},
+    },
+    layoutConfig = {},
+    availabilityConfig = {},
+  } = visualizations?.data?.userConfigs;
+
+  const valueForXSeries = valueOptions.dimensions
+    ? filterDataConfigParameter(valueOptions.dimensions)
+    : [];
+  const valueSeries = valueOptions.metrics ? filterDataConfigParameter(valueOptions.metrics) : [];
+  const boxOrientation = chartStyles.orientation || vis.orientation;
+  const isVertical = boxOrientation === vis.orientation;
+  const boxMode = chartStyles.boxMode || vis.boxmode;
+  let box;
```

Comment:
If this box is not used in between, we could use declare and assign value on line 84